### DISCUSSION TITLE
Special Attack Counter ToA Enhancement

### DISCUSSION THREAD

**Discussion by bennbatt** [runelite/runelite#17002] [Ideas]
Hi, pretty limited dev experience, but started poking around and familiarizing myself with the runelite code base.

What I'd like to try to implement is an enhancement to the Special Attack Counter plugin for Tombs of Amascut. Specifically, I'd like to have the thresholds automatically set for the npcs, based on the table on maximum defense reduction from the BGS. That table is found here: https://oldschool.runescape.wiki/w/Tombs_of_Amascut%2FStrategies#Equipment

Here are some things I've thought through so far:
- The config could be a boolean that overrides the threshold logic in a few places.
- I think the logic would need to apply for the case of weapon == BGS, and NPCs == Zebak, Kephri, Ba-Ba, etc.
- The current code grabs a static threshold set in plugin config, I think the getTextColor() and sendNotification() methods could take in a threshold arg instead to extend this.
- The IGNORED_NPCS would need at least NpcID.CORE and NpcID.ENERGY_SIPHON added... potentially more (thinking baba small monkeys and the Kephri room npcs too).
- P2 and P3 Warden logic might be a little bit tricky as there are two cases to solve for - namely P2 elidinis vs P2 tumeken.
- I don't know if logic around the instance you are in is possible (and sounds like a lot more effort) but that might be another avenue. Could tie in with the discussion around the Keris Partisan spec.

Would love to collaborate on something like this. I've done quite a bit of scripting and some basic NodeJS work before, but all self-taught and my understanding is pretty spotty.

**Comment by Nightfirecat** [comment]
I'm not entirely sure I understand what you're asking for. The plugin currently doesn't have thresholds of any kind--what would it be doing with these, recoloring the infobox text or something? How would it handle special attacks which do a percentage drain (like dwh or accursed sceptre) when we don't have a reliable way to know the monster's defense level?

**Reply by bennbatt** [reply]
![SpecialAttackCounter](https://github.com/runelite/runelite/assets/20671041/90b7fb42-91f2-4d84-b69c-4b5503ec2e4e)

This is the Special Attack Counter plugin, not the ToA plugin. You set thresholds for successful specs (or damage in the case of the BGS) and the plugin can notify you and/or change the info box from red to green against the NPC you're fighting. I use this for Corp for example, and I set the number of specs/damage I want for the kill.

What would be convenient is: while in Zebak room for example, override the threshold for the BGS counter and set it to 20 based on the maximum defense reduction table per the wiki. Then, when fighting the Obelisk, set the threshold to 40.

Agreed, other special attack weapons would be trickier to implement - I'm unsure if there is a way of consistently calculating defense level based on boss and invos.

**Reply by Nightfirecat** [reply]
This doesn't strike me as a ToA-specific enhancement, since the same thing would probably be useful/wanted at other monsters/other raids. That said, I have no idea how this would be implemented in a sane way.

**Reply by bennbatt** [reply]
Fair enough. My thinking would be a check box in special attack counter that overrides the Tombs bosses. As far as implementation goes, it looks tricky with how the current logic is setup in that plugin.
